docs/_source/docs/architecture.rst

```diff
@@ -135,15 +135,222 @@ to demonstrate ways of organising different projects.
+    # inherit these settings unless they explicitly override them. At minimum, you need the
+    # project_code and region, but you can set whatever StackGroup and Stack configurations you
+    # want inherited by the rest of the project.
+
+    # By making the project-code parameterized by environment name, it guarantees that one
+    # environment's stacks do not conflict with another environment's stacks.
+    project_code: indigo-{{ var.environment }}
+    region: {{ var.region }}
+
+    # Setting the template_bucket_name with a !stack_output means that every stack in the project
+    # (except the deployment stack) will have its template uploaded to that bucket. It ALSO means
+    # that every stack in the project will inherit a dependency on the deployment stack (except the
+    # deployment stack... see its comments below)
+    template_bucket_name: !stack_output project/deployment.yaml::BucketName
+
+    # By making the template_key_prefix parameterized by environment, it means that every
+    # environment's templates will be uploaded to its own directory structure within the template
+    # bucket, preventing conflicts.
+    template_key_prefix: {{ var.environment }}
+
+    # Setting the iam_role on the project like this means that all Sceptre actions on all stacks in
+    # the project will be done by assuming the deployment role whose ARN is output by the deployment
+    # stack (except for actions on the deployment stack... see its comments below).
+    iam_role: !stack_output project/deployment.yaml::DeploymentRoleArn
+
+
+The Deployment Stack
+^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+
+    # config/project/deployment.yaml
+
+    # This is a StackConfig for a stack that defines the project's basic requirements, that all
+    # stacks and all environments will depend on for deployment to work. Because the project config
+    # references this stack using !stack_output, all OTHER stacks in the project will depend upon
+    # this one.
+
+    # We override the project_code here so that there will only be ONE stack for deployment that is
+    # shared across all environments.
+    project_code: indigo-deployment
+
+    # Setting is_project_dependency has a few important effects for this stack
+    #   1. It disables dependencies. In other words, this stack cannot depend on any other stack.
+    #       This is how it bypasses the circular dependency that it would otherwise inherit from
+    #       the project's config (config/config.yaml)
+    #   2. It disables the !stack_output resolver, making it resolve to nothing. In other words, it
+    #       cannot inherit the template_bucket_name or iam_role from the project config. This is
+    #       good because this stack DEFINES the template bucket and iam_role.
+    is_project_dependency: True
+
+    # This template defines an S3 Bucket, outputting the created bucket as "BucketName" and an IAM
+    # role that can be assumed by the CI/CD system's role/user, outputting the ARN as
```

Comment:
where's the definition of the S3 bucket and IAM role?

Reply:
It's in the template. I didn't include the template in these docs. Are you suggesting I should include a sample CloudFormation template for these?

Reply:
Having a sample template would be useful so that someone could use it as a starting point.

To continue the point about being clunky from above having to define a custom template for this definitely does feel clunky to me, it makes the featuer feel like an advanced feature that an expert has to hand create. I wonder if we could provide some kind of minimal out of the box template that just works without the user having to create anything and they can "override" it by creating a file or something or parameters they can pass to modify the template.

Reply:
@zxiiro One of the things I suggested in #1114 was an auto-generation feature. I figured after all of these PRs got merged, I would start working on that. I know, for example, that `sam bootstrap` will auto-generate a stack that contains all the CloudFormation resources necessary to deploy the SAM template using a CI/CD pipeline. I really like that functionality and I think it could be implemented fairly simply with Sceptre.

Really, at minimum, that "bootstrapped" stack would just need to contain a simple s3 bucket for templates, optionally an IAM Role for deployment. It would actually be pretty simple.